tests/unity/example_test/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+#
```

Comment:
I'd like a .yml file or a .rst file that explains which boards I can expect this to work on.

Reply:
Also, which boards should this work on?

native_posix?

all nrf boards?

Reply:
I'm getting a build failure with nrf52840_pca10056:

```
Zephyr version: 1.14.0
-- Found PythonInterp: /home/sebo/bin/python3.4 (found suitable version "3.5.2", minimum required is "3.4") 
-- Selected BOARD nrf52840_pca10056
-- Loading /home/sebo/ncs/zephyr/boards/arm/nrf52840_pca10056/nrf52840_pca10056.dts as base
-- Overlaying /home/sebo/ncs/zephyr/dts/common/common.dts
nrf52840_pca10056.dts_compiled: Warning (simple_bus_reg): /soc/virtualcom: missing or empty reg/ranges property
Parsing Kconfig tree in /home/sebo/ncs/nrf/tests/unity/example_test/Kconfig
Loading /home/sebo/ncs/zephyr/boards/arm/nrf52840_pca10056/nrf52840_pca10056_defconfig as base
Merging /home/sebo/ncs/nrf/tests/unity/example_test/prj.conf
Configuration written to '/home/sebo/ncs/nrf/tests/unity/example_test/b/zephyr/.config'
-- Cache files will be written to: /home/sebo/.cache/zephyr
-- The C compiler identification is GNU 8.2.1
-- The CXX compiler identification is GNU 8.2.1
-- The ASM compiler identification is GNU
-- Found assembler: /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/arm-none-eabi-gcc
-- Performing Test toolchain_is_ok
-- Performing Test toolchain_is_ok - Success
Including module(s): nrf;nrfxlib
-- /usr/bin/ruby
-- RUBY_EXECUTABLE: /usr/bin/ruby
-- Generating test runner 
-- Generating cmock for header src/foo/foo.h
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sebo/ncs/nrf/tests/unity/example_test/b
[1/117] Preparing syscall dependency handling

[2/117] Generating include/generated/syscall_macros.h
[3/117] Generating misc/generated/syscalls_subdirs.trigger
[4/117] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h
[5/117] Generating include/generated/driver-validation.h
[6/117] Generating misc/generated/syscalls.json
[7/117] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[8/117] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
[9/117] Generating include/generated/offsets.h
[10/117] Building C object CMakeFiles/app.dir/src/foo/foo.c.obj
[11/117] Building C object CMakeFiles/app.dir/src/uut/uut.c.obj
[12/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj
[13/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj
[14/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj
[15/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj
[16/117] Building C object CMakeFiles/app.dir/runner/runner/runner_example_test.c.obj
FAILED: CMakeFiles/app.dir/runner/runner/runner_example_test.c.obj 
ccache /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/arm-none-eabi-gcc -DBUILD_VERSION=v1.14.0-rc1-896-g578dac80f7c3 -DKERNEL -DNRF52840_XXAA -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -Imocks -I../src -I.././src/uut -I../. -I/home/sebo/ncs/zephyr/kernel/include -I/home/sebo/ncs/zephyr/arch/arm/include -I/home/sebo/ncs/zephyr/include -I/home/sebo/ncs/zephyr/include/drivers -Izephyr/include/generated -I/home/sebo/ncs/zephyr/soc/arm/nordic_nrf/nrf52 -I/home/sebo/ncs/zephyr/soc/arm/nordic_nrf/include -I/home/sebo/ncs/zephyr/lib/libc/minimal/include -I/home/sebo/ncs/zephyr/ext/hal/cmsis/Include -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/drivers/include -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/hal -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/mdk -I/home/sebo/ncs/zephyr/ext/hal/nordic/. -I/home/sebo/ncs/zephyr/ext/debug/segger/rtt -I/home/sebo/ncs/nrf/include -I/home/sebo/ncs/zephyr/../cmock/vendor/unity/src -I/home/sebo/ncs/zephyr/../cmock/src -isystem /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/../lib/gcc/arm-none-eabi/8.2.1/include -isystem /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/../lib/gcc/arm-none-eabi/8.2.1/include-fixed -Os -nostdinc -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /home/sebo/ncs/nrf/tests/unity/example_test/b/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main -fno-common -mthumb -mcpu=cortex-m4 -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -Wpointer-arith -ffunction-sections -fdata-sections -mabi=aapcs -march=armv7e-m -std=c99 -MD -MT CMakeFiles/app.dir/runner/runner/runner_example_test.c.obj -MF CMakeFiles/app.dir/runner/runner/runner_example_test.c.obj.d -o CMakeFiles/app.dir/runner/runner/runner_example_test.c.obj   -c runner/runner/runner_example_test.c
In file included from /home/sebo/ncs/zephyr/../cmock/vendor/unity/src/unity.h:16,
                 from runner/runner/runner_example_test.c:29:
/home/sebo/ncs/zephyr/../cmock/vendor/unity/src/unity_internals.h:15:10: fatal error: setjmp.h: No such file or directory
 #include <setjmp.h>
          ^~~~~~~~~~
compilation terminated.
[17/117] Building C object CMakeFiles/app.dir/mocks/foo/mock_foo.c.obj
FAILED: CMakeFiles/app.dir/mocks/foo/mock_foo.c.obj 
ccache /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/arm-none-eabi-gcc -DBUILD_VERSION=v1.14.0-rc1-896-g578dac80f7c3 -DKERNEL -DNRF52840_XXAA -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -Imocks -I../src -I.././src/uut -I../. -I/home/sebo/ncs/zephyr/kernel/include -I/home/sebo/ncs/zephyr/arch/arm/include -I/home/sebo/ncs/zephyr/include -I/home/sebo/ncs/zephyr/include/drivers -Izephyr/include/generated -I/home/sebo/ncs/zephyr/soc/arm/nordic_nrf/nrf52 -I/home/sebo/ncs/zephyr/soc/arm/nordic_nrf/include -I/home/sebo/ncs/zephyr/lib/libc/minimal/include -I/home/sebo/ncs/zephyr/ext/hal/cmsis/Include -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/drivers/include -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/hal -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/mdk -I/home/sebo/ncs/zephyr/ext/hal/nordic/. -I/home/sebo/ncs/zephyr/ext/debug/segger/rtt -I/home/sebo/ncs/nrf/include -I/home/sebo/ncs/zephyr/../cmock/vendor/unity/src -I/home/sebo/ncs/zephyr/../cmock/src -isystem /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/../lib/gcc/arm-none-eabi/8.2.1/include -isystem /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/../lib/gcc/arm-none-eabi/8.2.1/include-fixed -Os -nostdinc -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /home/sebo/ncs/nrf/tests/unity/example_test/b/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main -fno-common -mthumb -mcpu=cortex-m4 -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -Wpointer-arith -ffunction-sections -fdata-sections -mabi=aapcs -march=armv7e-m -std=c99 -MD -MT CMakeFiles/app.dir/mocks/foo/mock_foo.c.obj -MF CMakeFiles/app.dir/mocks/foo/mock_foo.c.obj.d -o CMakeFiles/app.dir/mocks/foo/mock_foo.c.obj   -c mocks/foo/mock_foo.c
mocks/foo/mock_foo.c:4:10: fatal error: setjmp.h: No such file or directory
 #include <setjmp.h>
          ^~~~~~~~~~
compilation terminated.
[18/117] Building C object CMakeFiles/app.dir/src/example_test.c.obj
FAILED: CMakeFiles/app.dir/src/example_test.c.obj 
ccache /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/arm-none-eabi-gcc -DBUILD_VERSION=v1.14.0-rc1-896-g578dac80f7c3 -DKERNEL -DNRF52840_XXAA -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -Imocks -I../src -I.././src/uut -I../. -I/home/sebo/ncs/zephyr/kernel/include -I/home/sebo/ncs/zephyr/arch/arm/include -I/home/sebo/ncs/zephyr/include -I/home/sebo/ncs/zephyr/include/drivers -Izephyr/include/generated -I/home/sebo/ncs/zephyr/soc/arm/nordic_nrf/nrf52 -I/home/sebo/ncs/zephyr/soc/arm/nordic_nrf/include -I/home/sebo/ncs/zephyr/lib/libc/minimal/include -I/home/sebo/ncs/zephyr/ext/hal/cmsis/Include -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/drivers/include -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/hal -I/home/sebo/ncs/zephyr/ext/hal/nordic/nrfx/mdk -I/home/sebo/ncs/zephyr/ext/hal/nordic/. -I/home/sebo/ncs/zephyr/ext/debug/segger/rtt -I/home/sebo/ncs/nrf/include -I/home/sebo/ncs/zephyr/../cmock/vendor/unity/src -I/home/sebo/ncs/zephyr/../cmock/src -isystem /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/../lib/gcc/arm-none-eabi/8.2.1/include -isystem /home/sebo/Downloads/gnu-mcu-eclipse/arm-none-eabi-gcc/8.2.1-1.4-20190214-0604/bin/../lib/gcc/arm-none-eabi/8.2.1/include-fixed -Os -nostdinc -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /home/sebo/ncs/nrf/tests/unity/example_test/b/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main -fno-common -mthumb -mcpu=cortex-m4 -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -Wpointer-arith -ffunction-sections -fdata-sections -mabi=aapcs -march=armv7e-m -std=c99 -MD -MT CMakeFiles/app.dir/src/example_test.c.obj -MF CMakeFiles/app.dir/src/example_test.c.obj.d -o CMakeFiles/app.dir/src/example_test.c.obj   -c ../src/example_test.c
In file included from /home/sebo/ncs/zephyr/../cmock/vendor/unity/src/unity.h:16,
                 from ../src/example_test.c:6:
/home/sebo/ncs/zephyr/../cmock/vendor/unity/src/unity_internals.h:15:10: fatal error: setjmp.h: No such file or directory
 #include <setjmp.h>
          ^~~~~~~~~~
compilation terminated.
[19/117] Building C object zephyr/CMakeFiles/zephyr.dir/arch/common/sw_isr_common.c.obj
[20/117] Building C object zephyr/CMakeFiles/zephyr.dir/arch/common/isr_tables.c.obj
[21/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[22/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/work_q.c.obj
[23/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
[24/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/mempool.c.obj
[25/117] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj
ninja: build stopped: subcommand failed.

```

Reply:
currently it's native_posix. fixing `setjmp.h` error can come as incremental improvement. I'll add .rst

Reply:
Ah, k, got it working now :)

Reply:
Why was this resolved ?
I think what @SebastianBoe is asking for is a `sample.yaml` file in the folder:
`tests/unity/example_test`
with content similar to:
```
sample:
  name: example_test Sample
tests:
  test_build:
    build_only: true
    build_on_all: true
    platform_whitelist: native_posix
    tags: ci_build
```

Reply:
added file https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/175a1bd3f6799e9c1cb2e872d19df36d4b97c709/tests/unity/example_test/sample.yaml